Add Newbie Guide draft (#2)
BronzGreen-Honest-Power/LemiTree: Contributor_Guide.md

# Contributor Guide

## Table of Contents
1. Intro
2. How to contribute
3. Git & Github
	* Why we chose Git & Github?
	* What are some basic features of Git and Github?
	* How to use Git & Github?
4. Project structure
5. Markdown conventions
	* File name
	* "#" - Heading 1 - Name of the tactic
	* "##" - Heading 2 - Main sections
		* Intro
		* Why
		* How
		* Sources
		* Infographic, Video & Audio
	* "###" - Heading 3 - Subsections
	* "\[//\]: <> (BO-how)" - Comments - Scripting tools
	* Others
6. Project conventions
	* Tactic sources
		* Source types
		* Source format
	* Tactic classification
		* Frequency
		* Time
		* Expenses
		* Energy
		* Undefined

## Intro

> The LemiTree project is a database of self-development knowledge encompassing all different areas of life. To read more about our mission see [here](). [todo]

Reviewer: ```suggestion
The LemiTree project is a database of actionable self-development knowledge encompassing all essential areas of human life. To read more about our mission see [Welcome](Welcome.md).
```

This guide describes how you can concretely help us in our mission. By improving this toolbox, you will be contributing to the growth and flourishing of our fellow people. This way, more of us will be able to live our dreams and ambitions and live in harmony with our nature and our global ecosystem.

## How to contribute
There are currently 5 ways you may contribute to the project:
1. Adding new content or submitting changes to existing (especially incomplete) content

> 2. Reviewing existing or [incoming](https://github.com/BronzGreen-Honest-Power/LemiTree/pulls) contents, and suggesting improvements via comments

Reviewer: Great that you added these links here straight away!

In order to contribute directly (point 1 above), it is necessary to learn how to use Git (see below). We realise however, that Git may be challenging, and not everyone is dedicated enough, but would still like to contribute to content creation. Those people can do so indirectly (points 2 & 3) without much effort by simply making a Github account.

(In future updates we will simplify adding, changing and requesting new content. After those specific updates, we will make sure you do not have to understand Git in order to contribute.)

## Git & Github
Git & Github are simple but powerful tools for collaboration on a project. Their effectiveness is proven by thousands of software developers using them everyday.

### Why we chose Git & Github?
The first major benefit is that they enable us to have our toolbox available online while you and others recommend new content or changes. We can therefore easily work together with possibly thousands of contributors. Anyone who is inspired by our mission/project can thus choose to improve it. We don’t have to all know each other while working on our collective objective. This makes the project highly scalable.

> Github also allows our moderators to review these suggestions. When accepted, these changes will immediately show on the website. This makes sure that we only release the highest quality content.

Reviewer: ```suggestion
Github allows us to use scripts that enforce basic project conventions. This makes it easier to review contributions. It also enables a minimum standard of quality from the getgo. Github also allows our moderators to review the suggestions of others. When accepted, these changes will immediately show on the website. This makes sure that we only release the highest quality content.
```

Author: ```suggestion
Github allows us to use automation systems that enforce basic project conventions. This makes it easier to review contributions. It also enables a minimum standard of quality from the getgo. Github also allows our moderators to review the suggestions of others. When accepted, these changes will immediately show on the website. This makes sure that we only release the highest quality content.
```

Author: @wkoaber Your suggestion is not factually correct (Github doesn't allow for such automation on its own, but can be plugged in to other services that do). How about this instead?

Reviewer: Thank you for properly rereading it. Looks great, all for committing.

### How to use Git & Github?

[todo: cloning pulling branching staging committing pushing]

**[Github tutorial](https://www.edureka.co/blog/how-to-use-github/)**

## Project structure
In order to maintain consistency across the project, we follow some strict, but simple conventions. Simply put, all ***domains*** are broken down to smaller, plannable and actionable pieces. What we get as a result is a common hierarchical or tree-like structure.

On that tree, each node may branch out into many other "child-nodes". Such a complex node we call a ***subdomain*** or ***collection***. A child-node may also have its own children, thus expanding ad infinitum. In the database we achieve this hierarchy through nested directories (folders).

An atomic node - one that cannot be broken down - we call a ***tactic***. ***Tactics*** are represented as text files containing written descriptions and explanations of a specific set of actions that have some specific benefits.

> A ***tactic*** may have several integral subcomponents that are interdependent and cannot be separated, however if a ***tactic*** can be broken down, it should become a separate ***subdomain***.

Reviewer: ```suggestion
A ***tactic*** may have several integral subcomponents that are interdependent and cannot be separated. However, if a ***tactic*** can be broken down, it should become a separate ***subdomain***.
```

> Additionally, at each level of the hierarchy there shall be a tree graph (automatically generated) displaying local section of the tree up to two levels of depth (i.e. the selected ***subdomain*** and its child ***tactics*** and ***subdomains***, as well as *their* children), so that you can always see where you are, what is around you, and what awaits you one step deeper.

Reviewer: ```suggestion
Additionally, at each level of the hierarchy there shall be a tree graph (automatically generated). This graph displays the local section of the tree up to two levels of depth (i.e. the selected ***subdomain*** and its child ***tactics*** and ***subdomains***, as well as *their* children). This way, you can always see where you are, what is around you, and what awaits you one step deeper.
```

For an example, see [todo]

> DISCLAIMER: We do realise that the tree structure has its weaknesses, but overall we deem it to be useful.

Reviewer: ```suggestion
DISCLAIMER: We do realise that the tree structure has its weaknesses, but overall we deem it to be a good enough mix of realism and pragmatism.
```

## Markdown conventions

> To maintain consistency across the project, we had to decide on one specific format for our text files. Markdown seemed like the best candidate. Most notable reasons for that are: It’s simple, there are plenty of editors available, and it’s not bound by any specific operating system.

Reviewer: ```suggestion
To maintain consistency across the project, we had to decide on one specific format for our text files. Markdown seemed like the best candidate. Most notable reasons for that are:

- It’s simple, 
- There are plenty of editors available
- It’s not bound by any specific operating system.

```

All links should be included as **bullet points**. Since Markdown supports images, infographics can be formatted.

### "###" - Heading 3 - Subsections
Each of the three main sections may have subsections as described below.

* If a ***tactic***'s "How" section has several points to it, for example you need to perform different interdependent routines in the morning and in the evening, you can *optionally* use this heading to distinguish them within the same file. When describing these different instructions, we recommend to use **bullet points** for individual steps.
  Note: As explained above, if a ***tactic*** is complex, i.e. it can be broken down further into separate *independent* elements, it should be elevated to a separate ***subdomain***.
* Similarly, in the "Why" section, if there are multiple benefits, you can *optionally* list them using this heading.

> * We divide sources into several types (see below - *Project conventions: Tactic sources*). Each type has its own section, and each individual source should be a bullet point under the relevant section. Similarly to different intro formats, not all of these sections are mandatory (nor are they always possible) to fill out, however do not remove them. [todo]

Reviewer: ```suggestion
* We divide sources into several types (see below - *Project conventions: Tactic sources*). Each type has its own section, and each individual source should be a bullet point under the relevant section. Similarly to different contents formats, not all of these sections are mandatory (nor are they always possible) to fill out, however do not remove them. [todo]
```

Reviewer: What is the [todo] left here at the end?

Author: This section might need some changes once the sources in the template are finally settled.

Reviewer: Allright, clear!

### "\[//\]: <> (BO-how)" - Comments - Scripting tools

> As explained earlier, we might want to use various automation tools to reduce manual work. To make that easier, we may use Markdown's comment functionality to leave some "bookmarks" for locating various data points of interest. You don't need to worry about it too much, however if you see such comments, you should **not** (re)move them. Comments are **not** visible in the file preview. If you create a new tactic from scratch, you should use [the template](tactic_template.md), which already has comments for each section.

Reviewer: ```suggestion
As explained earlier, we might want to use various automation tools to reduce manual work. To make that easier, we may use Markdown's comment functionality to leave some "bookmarks" for locating various data points of interest. You don't need to worry about it too much, however if you see such comments, you should **not** (re)move them. Comments are **not** visible in the file preview. If you create a new tactic from scratch, you should use the [tactic template](tactic_template.md), which already has comments for each section.
```

(You don't need to know that, but if you're curious, "BO" means "beginning of", "EO" -- "end of".)

### Others
You're free to use any other features of Markdown as you see fit.

## Project conventions

### Tactic sources
#### Source types

> We trust the scientific method to provide the most accurate representation of reality. We realise, however, that it is somewhat limited. Many things in life are yet to be studied, many others can't be described with science altogether, and yet, to dismiss them would be foolish (e.g. financial ***tactics***).

Reviewer: ```suggestion
There are many methods of knowing. The scientific method however combines the strengths and counters some of the weaknesses of some of those methods. For specific and quantifiable domains, we trust the scientific method to provide the most accurate representation of reality. We realise, however, that it is somewhat limited. Many things in life are yet to be studied and many others can't be described with science altogether. To dismiss such domains and tactics would be foolish (e.g. financial ***tactics***).
```

> It is not up to us to decide other people's values. We have therefore decided that instead of restricting sources to purely scientific, we'd rather just ensure a clear distinction, and simply let the users decide what is convincing enough for them. Sources are therefore divided into several types:

Reviewer: ```suggestion
Further, it is not up to us to decide other people's values on what information they find reliable enough. We have therefore decided that instead of restricting sources to purely scientific, we'd rather just ensure a clear distinction. This way, the users can decide what is convincing enough for them. Sources are therefore divided into several types:
```

##### Scientific sources

> Scientific sources by definition need to apply the scientific method, however we'd also require for them to be somewhat supported by their respective scientific communities. This means, ideally, peer-reviewed articles from highly acclaimed journals. These kinds of sources are particularly relevant for areas like physical and mental health.

Reviewer: ```suggestion
Scientific sources by definition need to apply the scientific method. However, we'd also require them to be somewhat supported by their respective scientific communities. This means, ideally, peer-reviewed articles from highly acclaimed journals that are used in textbooks. This often portrays the current scientific consensus on a certain subject. These kinds of sources are particularly relevant for areas like physical and mental health.
```

##### Non-scientific sources
Articles or books that aren't strictly scientific or aren't yet peer reviewed. This category might be particularly relevant for topics such as philosophy, spirituality, business or relationships.

##### Expert opinion
Some topics might not have had any studies, but useful advice from experts in a field is often available in form of books, interviews or talks. For example, in business it might be worth listening to a billionaire, in sports -- to an olympic athlete, in writing -- to a best-selling author.

##### Other evidence

> Some areas may benefit highly from anecdotal evidence. Perhaps your grandpa has been happily married for 60 years and has some great relationship insights. Or maybe you yourself have discovered some useful tricks that might be worth sharing with the world.

Reviewer: ```suggestion
Some areas may benefit highly from anecdotal evidence. Perhaps your grandpa has been happily married for 60 years and has some great relationship insights. Or maybe you yourself have discovered some useful tricks that might be worth sharing with the world. Even the most accepted theories today used to be anecdotal perspectives.
```

Sometimes the status of a source might be disputable. This shall be decided by the consensus of the contributors. In cases where the disputes cannot reach an agreement, the final word belongs to us - the project maintainers.

#### Source format
We will accept various source formats: books (and e-books), articles (including web articles), documentaries, as well as interviews (including video and audio interviews), as long as they are relevant for the type of source they're applied to (i.e. interviews may serve as a source for "expert opinion", but not a "scientific source").

For consistency, we shall stick to a specific format of sources. We have chosen the bibliographical style of the *[Chicago Manual of Style](https://www.chicagomanualofstyle.org/tools_citationguide/citation-guide-1.html)*.

### Tactic classification

> While we wish for all ***tactics*** to be useful, they aren't necessarily equal - they vary in the quality of their benefits, as well as the kind and size of their costs. To make better decisions on prioritisation of those different ***tactics***, we have designed a system of classification that will allow for easy searching or sorting of the ***tactics***.

Reviewer: ```suggestion
While we wish for all ***tactics*** to be useful, they aren't necessarily equal - they vary in the quality of their benefits, as well as the kind and size of their costs. Their use might even change through time for you individually. To make better decisions on prioritisation of those different ***tactics***, we have designed a system of classification that will allow for easy searching or sorting of the ***tactics***.
```

We have also designed a special coding system for those classes, to make it easy for automation. These codes shall be in a comment at the top of each ***tactic*** file, before the title. Codes shall be separated by a comma, without spaces. For example:
>[//]: <> (FD2,EPA,TH2M30)

(The order of the codes doesn't matter but the order of symbols in each of the codes *does*.)

Keep in mind these classes are just estimations and not necessarily absolutes. Different kinds of exercises may affect different people differently, and products may change their price over time. Try to be accurate, but allow for some wiggle room.

#### Frequency
- F - frequency
Describes how often an action should be performed to achieve the benefits.
##### Intervals:
- D - days
- W - weeks
- M - months
- Q - quarters
- Y - years
- L - lifetime
##### Examples:
- FL (one timer action)
- FD (daily)
- FW (weekly)
- FW2 (bi-weekly)
- FM6 (every six months)

#### Time
- T - time
Describes how much time an action takes.
##### Quantity:
- H - hours
- M - minutes
##### Examples:
- TH2M30 (2 hours 30 minutes)
- TM5 (5 minutes)

#### Expenses
- X - expenses
Describes material, financial expenses.
##### Examples:

> - X100 ($100) [todo: tbd default currency]

Reviewer: ```suggestion
- X100 (€100)
```

Reviewer: We decided last time for now to keep the 'lingua franca' of currency, aka $ as the default currency
